### PR TITLE
fix: wrong snapshot names

### PR DESCRIPTION
**Pull request by sigus**
**Bugfix:** This PR addresses issue #6468

## Fix Details
This fix uses the pilot's name as a prefix for the new snapshot.

### CODE REVIEW

File: source/LoadPanel.cpp [hunk 1]
```diff
@@ -456,7 +456,7 @@ void LoadPanel::SnapshotCallback(const string &name)
 
 	// If a file with this name already exists, make sure the player
 	// actually wants to overwrite it.
-	string to = from.substr(0, from.size() - 4) + extension;
+	string to = Files::Saves() + selectedPilot + extension;
```
Comment (tehhowch) [inline source/LoadPanel.cpp:459]: Your proposed change is exactly equivalent to the current value based on my testing -- can you provide a step-by-step reproduction in which they will not be the same?

Comment (sigus) [reply source/LoadPanel.cpp:459]: 1. Open the load panel
2. Click the "Add snapshot" button
3. Type something (or leave the date there)
4. Click the "OK" button; then do not load the game
5. Click the "Add snapshot" button
6. Type something (or leave the date there)
7. Click the "OK" button

Comment (tehhowch) [reply source/LoadPanel.cpp:459]: That is the exact reproduction process I was performing. There is no difference between the original value and your proposed value.

I'm curious -- when you hover the snapshot created in step 4, what is the created timestamp? Is it the same as the source pilot? or different?

For me:
image
image
image

Comment (sigus) [reply source/LoadPanel.cpp:459]: > I'm curious -- when you hover the snapshot created in step 4, what is the created timestamp? Is it the same as the source pilot? or different?

It is later than original one.

Comment (tehhowch) [reply source/LoadPanel.cpp:459]: I think I would prefer to fix that issue -- `Files::Copy` not preserving the source files' timestamps -- than address only the names.

That has the side effect of fixing any other future uses of `Files::Copy` in the program to have the same behavior across platforms, and also results in a consistent ordering of the player's pilot file and snapshots across platforms.

**Comment by sigus** [comment]
Is support for macOS versions earlier than 10.13 intentionally preserved?

**Comment by tehhowch** [comment]
> Is support for macOS versions prior to 10.13 preserved intentionally?

Yes

**Comment by sigus** [comment]
This should be enough to fix #6468.

File: source/Files.cpp [hunk 2]
```diff
@@ -430,6 +431,18 @@ void Files::Copy(const string &from, const string &to)
 	CopyFileW(Utf8::ToUTF16(from).c_str(), Utf8::ToUTF16(to).c_str(), false);
 #else
 	Write(to, Read(from));
+	// Preserve the timestamps
```
Comment (quyykk) [inline source/Files.cpp:434]:
```suggestion
	// Preserve the timestamps of the original file.
```

File: source/Files.cpp [hunk 3]
```diff
@@ -430,6 +431,18 @@ void Files::Copy(const string &from, const string &to)
...
+	struct stat buf;
+	if(stat(from.c_str(), &buf))
+		LogError("Cannot stat \"" + from + "\"");
```
Comment (quyykk) [inline source/Files.cpp:437]:
```suggestion
		LogError("Error: Cannot stat \"" + from + "\".");
```

File: source/Files.cpp [hunk 4]
```diff
@@ -430,6 +431,18 @@ void Files::Copy(const string &from, const string &to)
...
+	else
+	{
+		struct utimbuf times;
+		times.actime = buf.st_atime;
+		times.modtime = buf.st_mtime;
+		if(utime(to.c_str(), &times))
+			LogError("Failed to preserve the timestamps for \"" + to + "\"");
```
Comment (quyykk) [inline source/Files.cpp:444]:
```suggestion
			LogError("Error: Failed to preserve the timestamps for \"" + to + "\".");
```

**Review by quyykk — APPROVED** [review]
Thank you! (Sorry that I messed up a bit)

**Pull request merged**